dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/RepositoryProducer.java

```diff
@@ -103,6 +121,39 @@ public void dispose(R repository) {
         handler.beanDisposed();
     }
 
+    /**
+     * Error handling for a timeout that occurs when attempting to obtain an
+     * EntityManagerBuilder.
+     *
+     * @param repositoryInterface the repository interface.
+     * @param exception           the TimeoutException.
+     * @return DataException with an appropriate error message.
+     */
+    @Trivial
+    private DataException excTimedOut(Class<?> repositoryInterface, Throwable exception) {
+        if (CheckpointPhase.getPhase() == CheckpointPhase.INACTIVE) {
```

Comment:
Question: Is it possible for the phase to be something other than `INACTIVE` after restore? 
The only reason I ask is that the CheckpointPhase has a method called `restored()` that checks both the phase and restored status: 
https://github.com/OpenLiberty/open-liberty/blob/32fda47ddb89df1d7d7f32861606bc28df1f9f38/dev/com.ibm.ws.kernel.boot.common/src/io/openliberty/checkpoint/spi/CheckpointPhase.java#L162-L171

I would assume that on restore the phase would always be INACTIVE, but maybe something to consider.